packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -7,22 +7,29 @@ exports.sourceNodes = (
   { boundActionCreators, getNode, hasNodeChanged, reporter },
   pluginOptions
 ) => {
-  const { createNode, deleteNode } = boundActionCreators
+  if (!(pluginOptions && pluginOptions.path)) {
+    reporter.panic(`
+You should pass 'path' options for gatsby-source-filesystem plugin
 
-  let ready = false
+See docs here - https://www.gatsbyjs.org/packages/gatsby-source-filesystem/
+    `)
+  }
 
   // Validate that the path exists.
   if (!fs.existsSync(pluginOptions.path)) {
-    console.log(`
+    reporter.panic(`
```

Comment:
This should still die after printing the error message.

Reply:
I guess `reporter.panic` does it for us:

https://github.com/gatsbyjs/gatsby/blob/78ad5b2c613b442d35239987d6c5e3892c54377e/packages/gatsby-cli/src/reporter/index.js#L25-L28

Reply:
Oh oops. Ok was confusing another discussion where we need a `panicOnBuild` where we shouldn't kill the development process but should kill the gatsby build process.